### ISSUE TITLE
Tribal Trouble Full Version

### ISSUE THREAD

**Issue by 0104hasan** [sunenielsen/tribaltrouble#25]
Hey, I'm following the steps to get the full version for tribal trouble but it is pretty confusing since I don't have a good computer science background. Would it be possible for someone to give me a simpler step by step instruction as to how I am supposed to get the full version. I am using windows by the way.

**Comment by veslav3** [comment]
@0104hasan I can help you using screenshare. Are you using Windows?

**Comment by 0104hasan** [comment]
Yes, I am using windows

**Comment by 0104hasan** [comment]
@veslav3 how do you want me to screen share

**Comment by veslav3** [comment]
@0104hasan https://meet.google.com/gkc-xfau-pja

**Comment by veslav3** [comment]
FYI for anyone reading this we shared screens. In case anyone needs help too, contact me.

**Comment by 0104hasan** [comment]
@veslav3 Do you know how to make multiplayer work. Like even if its a LAN connection.

**Comment by veslav3** [comment]
@0104hasan yes, you need some kind of server for this. With the current version you cannot do LAN.
I will work with my friend to get a server running.

**Comment by PurpleVoiddd** [comment]
Hey @veslav3 I would like to get my Tribal Trouble running on the full version as well. Could you be so kind & assist me as well? Thank you in advance!

**Comment by 0104hasan** [comment]
@PurpleVoiddd I could help you out. Its acctually really simple you only need to download two things. First download the installer, then download the registration file. Then, move the registration file to the tribal trouble file. -- When you download tribal trouble using the installer it automaticaly creates a file and shows you the location of the file. Just move the registraion file their and it shoudl work now.

**Comment by veslav3** [comment]
@PurpleVoiddd yes, I probably should make a video on how to do it.

**Comment by Fairuz-Afdhal** [comment]
For anyone reading this in the future, the README.md is not right about where do you put the registration file. Put it in your GAME folder instead of what the readme said.

**Comment by Gedehamsen021** [comment]
> @0104hasan yes, you need some kind of server for this. With the current version you cannot do LAN. I will work with my friend to get a server running.

hi, any news about server?, i tryed run the penguim team fork but server wont start, just giving the error

Could not find or load main class com.oddlabs.matchserver.MatchmakingServer

using this command

java -Djava.security.manager -Djava.security.policy=matchmaker.policy -server -ea -cp ../common/lib/java/mysql.jar:../common/lib/java/commons-pool-1.6.jar:../common/lib/java/commons-dbcp-1.4.jar:./build/matchmaking.jar com.oddlabs.matchserver.MatchmakingServer


the router server using command router works as well

java -Djava.security.manager -Djava.security.policy=router.policy -server -ea -cp ./build/router.jar com.oddlabs.routerserver.RouterServer

**Comment by team-penguin** [comment]
@Gedehamsen021 it looks like you haven't built the server project.
From within the server directory, run 'ant'.
Look at the contents of the build directory, there should be a matchmaking.jar, router.jar etc.
If that is the case, your server should start.

You also need to make sure that the DNS for oddlabs.com etc. is pointed at the loopback or network interface of your server.
The team-penguin50 fork has info on what your DNS should look like.
And....
You also have to have a database configured for the server to talk to. That info is at the aforementioned fork.

**Comment by Gedehamsen021** [comment]
> @Gedehamsen021 it looks like you haven't built the server project. From within the server directory, run 'ant'. Look at the contents of the build directory, there should be a matchmaking.jar, router.jar etc. If that is the case, your server should start.
> 
> You also need to make sure that the DNS for oddlabs.com etc. is pointed at the loopback or network interface of your server. The team-penguin50 fork has info on what your DNS should look like. And.... You also have to have a database configured for the server to talk to. That info is at the aforementioned fork.

what IDE is recommended?

**Comment by terraflops1048576** [comment]
@team-penguin You do not need to use DNS hacks; if you change the `settings` file in the same folder as the `registration` file, you can add these lines:

```
matchmaking_address=<your matchmaking server>
router_address=<your routing server>
```

**Comment by 0104hasan** [comment]
@veslav3 Hey Roy, 
I'm not sure if you remember but you helped me download the full version for Tribal Trouble a couple years back. 
I think last i spoke with you, you were saying something about creating a multiplayer server. Is their any update on that?

**Comment by veslav3** [comment]
@0104hasan yes well bad news in the sense that the efforts to bring it up again have been successful for a short time. It cost quite a lot of time and money for someone to do it.

**Comment by Gedehamsen021** [comment]
> @0104hasan yes well bad news in the sense that the efforts to bring it up again have been successful for a short time. It cost quite a lot of time and money for someone to do it.

Im working on a new version do the game, the multiplayer is done, now im working with spectator client for a tournament and a new ranking system

**Comment by 0104hasan** [comment]
@Gedehamsen021 what is the progress for that, would i be able to play that?

**Comment by ryan-linehan** [comment]
Hey @0104hasan, we have executables created so you don't have to do the whole thing to play now. Check out [tribaltrouble](https://tribaltrouble.org/) for executables for linux and windows (working on mac still) and the [fork](https://github.com/OmarAMokhtar/tribaltrouble)

There is also setup a discord (https://discord.gg/degeKcze) so that we can hopefully get some games scheduled since there aren't that many players yet. We're also discussing the dev work and such there as well. Happy to have others join us I'm trying to tag users on github that seem interested in different issues
